<launch>
  <arg name="rosbag_option" default="--clock"/>
  <arg name="bagfile_names" default="$(env HOME)/bagfiles/*"/>
  <arg name="manager" default="pcl_nodelet_manager" />

  <param name="use_sim_time" value="true" />

  <node pkg="rosbag" name="rosbag_play" type="play" args=" $(arg rosbag_option) $(arg bagfile_names)" output="screen" clear_params="true"/>


  <node pkg="nodelet" type="nodelet" name="$(arg manager)"
        args="manager" output="screen"/>

  <!-- <node pkg="nodelet" type="nodelet" name="pcd_writer" -->
  <!--       args="load pcl/PCDWriter $(arg manager)" -->
  <!--       clear_params="true"> -->
  <!--   <remap from="~input" to="/pcl_nodelet/normal_concat/output_pulse_out"/> -->
  <!-- </node> -->
  <node pkg="nodelet" type="nodelet"
        name="clustering"
        args="load jsk_pcl/EuclideanClustering $(arg manager)"
        output="screen" clear_params="true">
    <remap from="~input" to="/pcl_nodelet/normal_concat/output" />
    <rosparam>
      tolerance: 0.1
    </rosparam>
  </node>
  <node pkg="nodelet" type="nodelet"
        name="cluster_decomposer"
        args="load jsk_pcl/ClusterPointIndicesDecomposerZAxis $(arg manager)"
        output="screen" clear_params="true">
    <remap from="~input" to="/pcl_nodelet/normal_concat/output" />
    <remap from="~target" to="/clustering/output" />
    <rosparam>
    </rosparam>
  </node>

  <!-- <node pkg="aginika_ros_debug_tools" name="topic_pulse_drive" type="topic_pulse_drive" args="/cluster_decomposer/output00" /> -->


  <node pkg="aginika_pcl_ros" name="pointcloud_saver" type="selected_pointcloud_saver">
    <remap from="input" to="/cluster_decomposer/output00_pulse_out"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args=" -d $(find aginika_pcl_ros)/launch/rviz/extract_pcd_sample.rviz"/>

</launch>